backend/btrixcloud/zip.py

```diff
@@ -0,0 +1,147 @@
+"""
+Methods for interacting with zip/WACZ files
+"""
+import io
+import json
+import os
+import struct
+import zipfile
+import zlib
+
+from fastapi import HTTPException
+
+
+# ============================================================================
+EOCD_RECORD_SIZE = 22
+ZIP64_EOCD_RECORD_SIZE = 56
+ZIP64_EOCD_LOCATOR_SIZE = 20
+
+MAX_STANDARD_ZIP_SIZE = 4_294_967_295
+
+
+# ============================================================================
+async def extract_and_parse_log_file(client, bucket, key, log_zipinfo, cd_start):
+    """Return parsed JSON from extracted and uncompressed log"""
+    # pylint: disable=too-many-locals
+    file_head = await fetch(
+        client, bucket, key, cd_start + log_zipinfo.header_offset + 26, 4
+    )
+    name_len = parse_little_endian_to_int(file_head[0:2])
+    extra_len = parse_little_endian_to_int(file_head[2:4])
+
+    content = await fetch(
```

Comment:
Probably want to turn this into async generator all the way down, so could be something like:

```
body = await fetchBody(...)
if (gzip):
   return iter_lines(gzip_iter_chunks(body))
else:
   return body.iter_lines()
```

where:
```
async def gzip_iter_chunks(body):
  decompressor = zlib.decompressobj(-zlib.MAX_WBITS)
   async for chunk in body.iter_chunk():
       yield decompressor.decompress(chunk)
```
and iter_lines() just a copy of https://github.com/boto/botocore/blob/master/botocore/response.py#L135
which takes this generator instead..